lib/base-compiler.js

```diff
@@ -974,7 +994,7 @@ export class BaseCompiler {
         const inputFilename = path.join(dirPath, this.compileFilename);
         await fs.writeFile(inputFilename, source);
 
-        if (files) {
+        if (files && files.length > 0) {
```

Comment:
Is this a fix we should probably do in the main branch separately? (or is it incidental?)

Reply:
It's definitely a fix, but for some reason you only notice it with the new parser. I believe it might be a bug that "makes it right" in the current version.

Reply:
Got it. I've been doing far too much `python` recently, so I couldn't understand why you'd need both 😊 but you're right. Maybe consider making a super small change to main to fix it there, so this PR is more about enabling asm-parser than also healing previous wounds :)